### PR TITLE
Moved docker and scripts into tools. 

### PR DESCRIPTION
**Pull request by jespercedergren**
Added makefile to implement scrspts and updated READMEs. Created helpers to clean up infra in tests/helpers/infra.py used in conftest.py. Moved setup_infra.py into tests/setup. Skips api_s3_read tests. Minor import fixes when import patches.

### CODE REVIEW

File: tools/docker/docker_images/test/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM spark_hadoop_python_base AS test_base
+#FROM spark_hadoop_python_base AS test_base
```
Comment (Thelin90) [inline tools/docker/docker_images/test/Dockerfile:1]: Remove comment

File: tools/scripts/docker_build_test_base.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+docker build -t python_base ./tools/docker/docker_images/python_base
+docker build -t python_spark_hadoop_base ./tools/docker/docker_images/python_spark_hadoop_base
+#docker build -t spark_hadoop_python_base ./tools/docker/docker_images/spark_hadoop_python_base
```
Comment (Thelin90) [inline tools/scripts/docker_build_test_base.sh:3]: Remove comment

File: tests/conftest.py [hunk 3]
```diff
@@ -176,16 +173,16 @@ def spark_session_minio():
...
 
-    hadoop_conf.set("spark.shuffle.service.enabled",  "true")
+    hadoop_conf.set("spark.shuffle.service.enabled", "true")
     hadoop_conf.set("fs.s3a.fast.upload", "true")
-    #hadoop_conf.set("spark.sql.parquet.cacheMetadata", "true")
+    # hadoop_conf.set("spark.sql.parquet.cacheMetadata", "true")
 
     yield spark
-    print('Delete Spark session...')
+    print("Delete Spark session...")
```
[the remark this answers is not in the document]
Comment (Thelin90) [reply tests/conftest.py:185]: Also will this print even ever happen? It prints after the yield? It should be:

```python
logging.info("Detele SparkSession")
yield spark
```

?

File: tests/integration_tests/api/test_api_read_s3.py [hunk 4]
```diff
@@ -1,25 +1,31 @@
...
+import pytest
 import requests
 
+from tests.config import server_service
 from tests.helpers.common import pd_equals
 from tests.helpers.io import save_parquet
-from tests.config import server_service
 
 
+# TODO fix and unskip tests
```
[the remark this answers is not in the document]
Comment (jespercedergren) [reply tests/integration_tests/api/test_api_read_s3.py:12]: Isn't this the way to to write a "TODO"?

File: tests/helpers/infra.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import boto3
+
+
+def cleanup_buckets(s3_resource: boto3.resource):
```
[the remark this answers is not in the document]
Comment (jespercedergren) [reply tests/helpers/infra.py:4]: This was deprecated so I'm using 
```from mypy_boto3 import s3```

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+setup_infra:
+	tools/scripts/setup_infra.sh
+
+.PHONY: run_tests_clean
+run_tests_clean:
+	tools/scripts/run_tests_clean.sh ${test}
+
+.PHONY: run_tests
+run_tests:
+	tools/scripts/run_tests.sh ${test}
```
[the remark this answers is not in the document]
Comment (jespercedergren) [reply Makefile:55]: This is intended to test specific tests. Could have this approach for the run_tests_clean above.

**Pull request merged**